irispy/iris_tools.py

```diff
@@ -212,6 +220,88 @@ def get_iris_response(pre_launch=False, response_file=None, response_version=Non
     return iris_response
 
 
+def fit_iris_xput(time_obs, time_cal_coeffs, cal_coeffs):
+    """
+    To calculate the coefficients of best-fit time function for throughput,
+    for which there are two modes:
+    1. Perform fit: supply xput and single element ``cal_coeffs``.
+    2. Apply fit: supply full ``cal_coeffs``.
+
+    Parameters
+    ----------
+    time_obs: a numpy array of floats
+    - Times of datapoints.
+
+    time_cal_coeffs: a numpy array of floats
+    - Start and end times of intervals of constant ``cal_coeffs[i]``.
+
+    cal_coeffs: a numpy array of floats
+    - Coefficients of best-fit function.
+
+    Returns
+    -------
+    Yields the values of fit for times ``time_obs``.
+
+    """
+    if isinstance(time_cal_coeffs, np.ndarray):
+        if isinstance(cal_coeffs, np.ndarray):
+            size_time_cal_coeffs = time_cal_coeffs.shape
+            size_cal_coeffs = cal_coeffs.shape
+
+    if size_time_cal_coeffs[0] != 2 or size_cal_coeffs[0] < 2:
+        # Raise ValueError as time coefficient have the wrong format.
+        raise ValueError("Incorrect number of elements either in time_cal_coeffs or in cal_coeffs.")
+        # Exponent for transition between exp.decay intervals.
+        transition_exp = 1.5
```

Comment:
All lines from this one and below should not be unindented four spaces to take them out of the above ```if``` statement part of this ```if``` statement.  As it is, the code from here on will never be executed.